components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/dao/impl/DeviceDAOImpl.java

```diff
@@ -330,4 +330,58 @@ public int getDeviceCount() throws DeviceManagementDAOException {
         return deviceCount;
     }
 
+    /**
+     * Get the list of devices that matches with the given device name.
+     *
+     * @param deviceName Name of the device.
+     * @param tenantId
+     * @return device list
+     * @throws DeviceManagementDAOException
+     */
+    @Override
+    public List<Device> getDevicesByName(String deviceName, int tenantId) throws DeviceManagementDAOException {
+        Connection conn = this.getConnection();
+        PreparedStatement stmt = null;
```

Comment:
The 'stmt' variable should be initialized prior to passing to 'cleanupResources(stmt, resultset)' method.